pages/api/v1/contents/index.public.js

```diff
@@ -48,11 +51,9 @@ async function getHandler(request, response) {
   const results = await content.findWithStrategy({
     strategy: request.query.strategy,
     where: {
-      parent_id: null,
+      parent_id: request.query.with_children ? undefined : null,
       status: 'published',
-    },
-    attributes: {
-      exclude: ['body'],
+      $not_null: request.query.with_root === false ? ['parent_id'] : undefined,
     },
     page: request.query.page,
```

Comment:
O que vocês acham de não obter o `body` se não for necessário?

```suggestion
    attributes: {
      exclude: request.query.with_children ? undefined : ['body'],
    },
    page: request.query.page,
```

Reply:
Só faltou ignorar o loop para o caso `!request.query.with_children`

https://github.com/filipedeschamps/tabnews.com.br/blob/6d9676c1ab6b285ae0b196aa9b2727e7eb5c0fca/pages/api/v1/contents/index.public.js#L69-L75

Reply:
> para o caso `!request.query.with_children`

@aprendendofelipe Vai falhar no teste `get root and children content with "with_children=true` onde `with_children=true` precisa do body e o root não, como o padrão é retornar o root content sem "with_root", acaba retornando body no root content por que não fez o loop.

Reply:
O loop só precisa ser executado justamente quando `with_children=true`:

```js
 if (request.query.with_children) {
   for (const content of secureOutputValues) { 
     if (content.parent_id) { 
       content.body = removeMarkdown(content.body, { maxLength: 255 }); 
     } else { 
       delete content.body; 
     } 
   }
 }

Reply:
> O loop só precisa ser executado justamente quando `with_children=true`:
> 
> ```js
>  if (request.query.with_children) {
>    for (const content of secureOutputValues) { 
>      if (content.parent_id) { 
>        content.body = removeMarkdown(content.body, { maxLength: 255 }); 
>      } else { 
>        delete content.body; 
>      } 
>    }
>  }
> ```

@aprendendofelipe sim, faz sentindo agora.